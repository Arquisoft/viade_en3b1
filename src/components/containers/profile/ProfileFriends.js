import React, { useEffect, useState } from 'react';
import { Container, Typography, makeStyles, CssBaseline, Grid, Paper, Card, Avatar, CardContent, Link, CardHeader, IconButton } from '@material-ui/core';
import NavBar from '../../ui/main/NavBar';
import Footer from '../../ui/main/Footer';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import cache from '../../../cache/UserCache';
import LinkIcon from '@material-ui/icons/Link';
import UserCache from '../../../cache/UserCache';
import { GetUserWebId } from '../../../parser/UserDataHandler';

// var friendsList = [];

export default function ProfileFriends() {
    const classes = useStyles();

    return (
        <div className={classes.root}>
            <NavBar />
            <CssBaseline />
            <Container
                component="main"
                className={classes.main}
                maxWidth="md"
            >
                <Typography variant="h3">Your Profile</Typography>

                <Paper
                    elevation={0}
                    className={classes.paper}>
                    <Tabs
                        value={1}
                        indicatorColor="primary"
                        textColor="primary"
                    >
                        <Tab selected={false} label="General" href="#/profile" />
                        <Tab selected={true} label="Friends" href="#/profile/friends" />
                    </Tabs>
                </Paper>

                <FriendCardList />
            </Container>

            <Footer />
        </div>
    )
}

function FriendCardList() {
    const classes = useStyles();

    // #### WEBID ####
    const [webId, setWebId] = useState(0);

    GetUserWebId().then((url) => {
        setWebId(url);
    });

    // #### FRIENDS ####
    const [friendsList, setFriendsList] = useState([]);
    useEffect(() => {
        // (async function () {
        //     const list = cache.getFriends();
        //     setFriendsList(list);
        // })();
        // cache.getFriends().then((list) =>{
        //     setFriendsList(list);
        // });
        const list = cache.getFriends();
        setFriendsList(list);
        console.log(list);

    }, []);

    if (!friendsList.length) {
        return (
            <div className={classes.friendsList}>
                <Typography variant="h6">
                    Oops, it seems you don't have any friends yet.
                </Typography>
                <Typography variant="subtitle1">
                    You can add new friends in <Link style={{ color: "#7c4dff" }} target="_blank" href={webId}>your profile</Link>.
                </Typography>

            </div>
        );
    }

    return (
        <Grid
            container
            spacing={2}
            alignItems="center"
            justify="center"
            className={classes.friends}
        >
            {friendsList.map((each, index) => {
                return (
                    <Grid item key={index}>
                        <FriendCard friend={each} />
                    </Grid>
                )
            })}

        </Grid>
    );
}

function FriendCard(props) {
    const classes = useStyles();

    const { friend } = props;

    var photo = friend.getPhoto();
    var name = friend.getName();
    var webid = friend.getWebId();

    return (
        <Card variant="outlined" className={classes.fCard}>
            <CardHeader
                avatar={
                    <Avatar src={photo} className={classes.fPhoto} />
                }
                action={
                    <IconButton aria-label="go to solid">
                        <Link style={{ color: "#7c4dff" }} target="_blank" href={webid}>
                            <LinkIcon />
                        </Link>
                    </IconButton>
                }
            />
            <CardContent style={{ textAlign: 'center' }}>
                <Typography variant="h6" color="textSecondary" component="p">
                    {name}
                </Typography>
            </CardContent>
        </Card>
    );
}

const useStyles = makeStyles((theme) => ({
    root: {
        display: 'flex',
        flexDirection: 'column',
    },
    main: {
        marginTop: theme.spacing(8),
        marginBottom: theme.spacing(2),
    },
    paper: {
        borderBottom: '1px solid #e8e8e8',
        marginTop: theme.spacing(2),
        backgroundColor: "#fafafa",
        width: '100%'
    },
    friends: {
        marginTop: theme.spacing(5),
    },
    fPhoto: {
        width: theme.spacing(13),
        height: theme.spacing(13),
        marginLeft: '3rem',
        marginTop: '1rem'
    },
    fCard: {
        display: 'block',
        width: '15rem',
        height: theme.spacing(30)
    },
    friendsList: {
        textAlign: 'center',
        marginTop: theme.spacing(7),
        marginBottom: theme.spacing(50)
    }
}));
