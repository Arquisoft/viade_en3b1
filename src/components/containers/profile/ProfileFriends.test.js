import ProfileFriends from './ProfileFriends';

test('Define ProfileFriends', () => {
  expect(ProfileFriends).toBeDefined();
});

test('Renders correctly', () => {
  const wrapper = render(<ProfileFriends />);
  expect(wrapper).toMatchSnapshot();
});