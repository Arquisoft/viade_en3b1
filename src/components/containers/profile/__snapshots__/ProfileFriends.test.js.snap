// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  class="makeStyles-root-1"
>
  <div
    class="makeStyles-root-8"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-9 MuiToolbar-gutters"
      >
        <div
          class="MuiGrid-root makeStyles-link-11"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorInherit"
              href="#/home"
            >
              <div
                class="MuiGrid-root MuiGrid-container"
              >
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <img
                    alt="Viade Logo"
                    src="logo_2.svg"
                    style="height:2.5rem;width:2.5rem"
                  />
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <h4
                    class="MuiTypography-root MuiTypography-h4"
                  >
                    ViaDe
                  </h4>
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
    </header>
  </div>
  <main
    class="MuiContainer-root makeStyles-main-2 MuiContainer-maxWidthMd"
  >
    <h3
      class="MuiTypography-root MuiTypography-h3"
    >
      Your Friends
    </h3>
    <div
      class="MuiPaper-root makeStyles-paper-3 MuiPaper-elevation0 MuiPaper-rounded"
    >
      <div
        class="MuiTabs-root"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed"
          style="overflow:hidden"
        >
          <div
            class="MuiTabs-flexContainer"
            role="tablist"
          >
            <a
              aria-disabled="false"
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary"
              href="#/profile"
              role="tab"
              tabindex="0"
            >
              <span
                class="MuiTab-wrapper"
              >
                General
              </span>
            </a>
            <a
              aria-disabled="false"
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected"
              href="#/profile/friends"
              role="tab"
              tabindex="0"
            >
              <span
                class="MuiTab-wrapper"
              >
                Friends
              </span>
              <span
                class="PrivateTabIndicator-root-226 PrivateTabIndicator-colorPrimary-227 MuiTabs-indicator"
              />
            </a>
            <a
              aria-disabled="false"
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary"
              href="#/profile/groups"
              role="tab"
              tabindex="0"
            >
              <span
                class="MuiTab-wrapper"
              >
                Groups
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="makeStyles-friendsList-7"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6"
      >
        Oops, it seems you don't have any friends yet.
      </h6>
      <h6
        class="MuiTypography-root MuiTypography-subtitle1"
      >
        You can add new friends in your 
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
          href="0"
          style="color:#7c4dff"
          target="_blank"
        >
          Solid profile
        </a>
        .
      </h6>
    </div>
  </main>
  <div>
    <footer
      class="makeStyles-footer-232"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthXs"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6"
            >
              ViaDe En3B1
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
            >
              Made by 
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href="https://www.linkedin.com/in/pablocanalsuarez/"
                style="color:#302c58"
                target="_blank"
              >
                Pablo Ca√±al
              </a>
            </h6>
            <div>
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href="https://www.linkedin.com/in/pablocanalsuarez/"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
                  />
                </svg>
              </a>
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href="mailto:uo264903@uniovi.es"
                style="margin-left:0.5rem"
                target="_blank"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
                  />
                </svg>
              </a>
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-flexItem MuiDivider-vertical"
          />
          <div
            class="MuiGrid-root MuiGrid-item"
            style="margin-top:1rem"
          >
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              href="https://github.com/Arquisoft/viade_en3b1/"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
